src/hooks/useBreakpoint.tsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+import { useDebouncedCallback } from "use-debounce";
+import { BreakpointKey, Breakpoints } from "../Css";
+
+/**
+ * A React hook to return a record of responsive breakpoints that updates on resize.
+ *
+ * @example
+ * const { breakpoints } = useBreakpoint();
+ * if(breakpoints.mdAndDown) {...do something cool}
+ */
+
+type BreakpointsType = Partial<Record<BreakpointKey, boolean>>;
+
+export function useBreakpoint(): { breakpoints: BreakpointsType } {
+  const [breakpoints, setBreakpoints] = useState(matchMediaBreakpoints());
+
+  useEffect(() => {
+    const handleResize = useDebouncedCallback(
```

Comment:
I don't think you can use a hook inside a hook like this. You might need to put this outside of the useEffect